account_reconcile_restrict_partner_mismatch/models/account_move_line.py

```diff
@@ -0,0 +1,34 @@
+# -*- coding: utf-8 -*-
+# Copyright 2019 Camptocamp SA
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl.html).
+
+from odoo import api, models, _
+from odoo.exceptions import UserError
+from odoo.tools import config
+
+
+class AccountMoveLine(models.Model):
+    _inherit = "account.move.line"
+
+    @api.multi
+    def reconcile(self, writeoff_acc_id=False, writeoff_journal_id=False):
+        if (config['test_enable']
+                and not self.env.context.get('specific_tests')):
```

Comment:
```suggestion
                and not self.env.context.get('test_partner_mismatch')):
```
Please use a more 'specific' context key :wink: